### PR TITLE
Provena Release v2.0

### PR DESCRIPTION
**Pull request by PeterBaker0**
# Provena Release v2.0

## Major Features

- Dataset Release Workflow
- Item and Dataset Versioning
- Generic Asynchronous Job Infrastructure (codename GAJI)
- Additional Prov Queries
- System version API endpoint

## Acknowledgements

Development

- @PeterBaker0
- @xinyuhoulucy  
- @Rosspet

Review

- @jyucsiro 
- @fre171csiro 
- @LindaThomas009 

## Migrations

TBD

### Includes the following PRs

-   RRAPIS-1458 (Major): Dataset Approval Workflow API Implementation (#36)
-   RRAPIS-1485 (Minor): Differentiate-versions-in-all-searching-forms (#35)
-   RRAPIS-1470 (Minor): Rename "jobs-api" to "job-api" including domain route (#33)
-   RRAPIS-1480 (Bug Fix): Fix dataset revision data access backdoor through revert operation (#31)
-   RRAPIS-1481 (Minor): Adds additional clean-up items to integration tests to catch generated Create and Version activities (#30)
-   RRAPIS-1476 (Bug Fix): GAJI bug when launching task during shutdown phase (#29)
-   RRAPIS-1483 (Minor): Change all "Revision" to "Version" (#28)
-   RRAPIS-1479 (Minor): Bump node to 20 for general use and 18 for CodeBuild
-   RRAPIS-1457 (Bug Fix): Err message not being shown properly (#18)
-   RRAPIS-1441 (Major): Versioning-frontends (#22)
-   RRAPIS-1475 (Minor): GAJI job microservice template + email service + job wake up + Job UI admin mode (#25)
-   RRAPIS-1442 (Major): Registry item versioning backend impl. (incl. Datasets) (#20)
-   RRAPIS-1452(Bug Fix): Duplicate resource headers fix (#15)
-   RRAPIS-1431(Minor): Creation of Quick Deploy Pipeline (#16)
-   RRAPIS-1349(Major): Registry Version Info Endpoint
-   RRAPIS-1445 (Major): Generic Asynchronous Job Infrastructure (GAJI)
-   RRAPIS-1312 (Minor): Filters-for-prov-graph-UI (#11)
-   RRAPIS-1339 (Minor): Model Run Display Name (#12)
-   RRAPIS-643 (Minor): Add-prov-graph-to-dataset-view (#8)
-   RRAPIS-1338 (Bug Fix): Bug-when-updating-externally-reposited-metadata (#9)
-   RRAPIS-1433 (Bug Fix): Fix-layout-bugs (#4)
-   RRAPIS-1375 (Minor): Make-registered-entries-boxes-overflow (#3)
-   RRAPIS-1319(Bug Fix): Prov-graph-does-not-load-first-time (#308)

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "editor.autoIndent": "keep",
     "editor.tabSize": 4,
     "autoDocstring.docstringFormat": "numpy",
-    "autoDocstring.customTemplatePath": "templates/docstrings/rrap_docstring.mustache",
```
Comment (jyucsiro) [inline .vscode/settings.json:9]: Not sure if this is used?

**Review by jyucsiro — CHANGES_REQUESTED** [review]
Scanned the files and looked for RRAP. Found a few references and noted it down. is it possible to do a full install of this branch separately for testing purposes?

File: infrastructure/README.md [hunk 2]
```diff
@@ -150,7 +150,7 @@ The Provena application has a mixture of
 
 For this reason, we provide a rich configuration interface into the application.
 
-Below, we will provide an overview of some of the key concepts for managing Provena configuration and interacting with your deployed application. 
+Below, we will provide an overview of some of the key concepts for managing Provena configuration and interacting with your deployed application.
 
 **We recommend that new users, who want to deploy Provena, visit the more thorough deployment guide available here: [Provena from Scratch](https://confluence.csiro.au/display/RRAPIS/HOW+TO+Deploy+Provena+from+SCRATCH).**
```
[the remark this answers is not in the document]
Comment (PeterBaker0) [reply infrastructure/README.md:155]: yep - let's make a separate ticket for it?

File: utilities/packages/typescript/react-libs/src/themes/rrap/scssConfig.module.scss
[the remark this answers is not in the document]
Comment (PeterBaker0) [reply utilities/packages/typescript/react-libs/src/themes/rrap/scssConfig.module.scss:1]: It was already in there - but I agree it doesn't need to be , though it is handy to see a reference. 

I don't mind either way

File: .vscode/settings.json (hunk 1 above)
Comment (PeterBaker0) [reply .vscode/settings.json:9]: Agreed - I'll remove it

File: data-store-api/config.py [hunk 3]
```diff
@@ -41,6 +42,9 @@ class Config(BaseConfig):
     S3_STORAGE_BUCKET_NAME: str
     HANDLE_API_ENDPOINT: str
     REGISTRY_API_ENDPOINT: str
+    job_api_endpoint: str
```
Comment (xinyuhoulucy) [inline data-store-api/config.py:45]: Does this "job_api_endpoint" need to be in all capital format?

File: data-store-api/tests/test_authorization.py [hunk 4]
```diff
@@ -8,8 +8,11 @@
     HANDLE_API_ENDPOINT="",
     # these calls need to be mocked
     REGISTRY_API_ENDPOINT="",
+    job_api_endpoint="",
```
Comment (xinyuhoulucy) [inline data-store-api/tests/test_authorization.py:11]: same here

File: data-store-api/tests/test_functionality.py [hunk 5]
```diff
@@ -31,9 +32,12 @@ def test_config() -> Config:
         HANDLE_API_ENDPOINT="",
         # these calls need to be mocked
         REGISTRY_API_ENDPOINT="",
+        job_api_endpoint="",
```
Comment (xinyuhoulucy) [inline data-store-api/tests/test_functionality.py:35]: same here

File: data-store-ui/.env.example
Comment (xinyuhoulucy) [inline data-store-ui/.env.example:1]: my bad, missed adding VITE_THEME_ID="${YOUR_THEME_ID}" here, maybe the same is missing for the other three UIs in the .env.example file as well

**Review by xinyuhoulucy — CHANGES_REQUESTED** [review]

File: infrastructure/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jyucsiro) [reply infrastructure/README.md:155]: ok

File: utilities/packages/typescript/react-libs/src/themes/rrap/scssConfig.module.scss
[the remark this answers is not in the document]
Comment (jyucsiro) [reply utilities/packages/typescript/react-libs/src/themes/rrap/scssConfig.module.scss:1]: ok noted. let's leave it then

File: data-store-ui/.env.example
Comment (jyucsiro) [reply data-store-ui/.env.example:1]: ok - will you @xinyuhoulucy add them and include that in this PR?

Comment (xinyuhoulucy) [reply data-store-ui/.env.example:1]: yep, I will do that :)

File: data-store-api/config.py (hunk 3 above)
Comment (PeterBaker0) [reply data-store-api/config.py:45]: If anything, I'd prefer to change all of the others to lower case. That's the standard being used in most of the other APIs. 

Let's make this a separate issue to unify this across all the APIs - I know there's a mixture in other areas as well.

File: data-store-api/tests/test_authorization.py (hunk 4 above)
Comment (PeterBaker0) [reply data-store-api/tests/test_authorization.py:11]: Same as above

File: data-store-ui/.env.example
Comment (PeterBaker0) [reply data-store-ui/.env.example:1]: I think pushing straight here SHOULD be okay - as long as we can merge back into our upstream once done - just give it a go and we can resolve conflicts if required.

**Review by xinyuhoulucy — APPROVED** [review]

**Review by jyucsiro — APPROVED** [review]

**Pull request merged**